Sources/Views/MessageLabel.swift

```diff
@@ -322,6 +353,36 @@ open class MessageLabel: UILabel {
         return results
     }
 
+    /**
+     Simply check if the detector type is a .custom
+     All .custom contains an `NSRegularExpression`
+
+     - Parameters: detector: `DetectorType` that you want to execute
+
+     - Returns: Bool true if the detector type is a .custom, otherwise false
+     */
+    private func isCustomDetector(for detector: DetectorType) -> Bool {
+        if case .custom = detector { return true }
+        return false
+    }
+
+    /**
+     Take a custom detector and apply the matching to the text
+     
+     - Parameters: detector: `DetectorType` that you want to execute
+     - Parameters: text: where we apply the regular expression
+     
+     - Returns: an array of `NSTextCheckingResult` that contains all maching for this detector or nil if it fails
+     */
+    private func parse(custom detector: DetectorType, for text: NSAttributedString, in range: NSRange) -> [NSTextCheckingResult]? {
+        switch detector {
+        case .custom(let regex):
+            return regex.matches(in: text.string, options: [], range: range)
+        default:
+            return nil
```

Comment:
I think this should be a `fatalError()` as the method is only to be used with custom detector types

Reply:
Nice idea @nathantannar4